### DISCUSSION TITLE
View All Excel-Style Filter(s) of a Data Grid At-A-Glance

### DISCUSSION THREAD

**Discussion by SkylerLutz** [IgniteUI/igniteui-angular#13516] [Q&A] [answered]
## Question  
Hi, I'm using Excel-Style Filtering in a Data Grid in a production application. My users are complaining that if they filter on multiple columns, they don't have a way to view all filters applied to the grid at a glance, and this leads to confusion about what records are actually presented. Is there a way to present the user with some sort of aggregate/summary of all filters, so that users can understand at a glance what filters are applied? Thanks

**Comment by ddaribo** [comment, marked as answer]
Hi @SkylerLutz,

There is not a built-in UI for this in the grid, nevertheless it could easily be achieved by leveraging the exposed API. The [filteringExpressionsTree]( https://www.infragistics.com/products/ignite-ui-angular/docs/typescript/latest/classes/IgxGridComponent.html#filteringExpressionsTree) property contains information about the applied filters. Additionally, the [filteringExpressionsTreeChange](https://www.infragistics.com/products/ignite-ui-angular/docs/typescript/latest/classes/IgxGridComponent.html#filteringExpressionsTreeChange) event is fired each time the filters are changed.

As the name suggests, the structure of the object is recursive, which is particularly helpful for the Advanced filtering scenario. However, for the Excel Style Filtering case, the hierarchy goes as deep as two levels, so it is straightforward to extract the relevant info and construct a data object. 
How this information is presented to the user is a matter of personal preference. To demonstrate a possible approach, I created [this sample](https://stackblitz.com/edit/igx-grid-esf-visualize-all-filtering-expressions?file=src%2Fapp%2Fgrid%2Fgrid-excel-style-filtering-sample-1%2Fgrid-excel-style-filtering-sample-1.component.ts), where a button is included in the grid toolbar, which opens a dialog with an hierarchical grid. This grid presents information about the current filtering expressions. It is hierarchical, because, in the case of many selected column values, the “search value” is actually a set and can be visually presented in a child table for each record:

![esf-hierarchy](https://github.com/IgniteUI/igniteui-angular/assets/35433937/79531346-25ce-44ea-a7c4-08d43605303b)

Please, check out the sample and let me know if it helps achieving your requirement.
